### PR TITLE
Add linkcheck to tox + CI + fix urls

### PR DESCRIPTION
**Pull request by nabobalis**
Some NOAA XRS pages should work but are broken even when loading a NOAA site and pressing the guide url on the page.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -10,11 +10,13 @@ envlist =
     build_docs{,-gallery}
     codestyle
     core_deps
+    linkcheck
 
 [testenv]
 pypi_filter = file://.test_package_pins.txt
 allowlist_externals=
     /bin/sh
+    sphinx-build
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply tox.ini:19]: It errored for me locally.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -79,6 +79,7 @@ jobs:
       envs: |
         - linux: build_docs-gallery
           pytest: false
+          coverage: ''
```
Comment (nabobalis) [inline .github/workflows/ci.yml:82]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -103,14 +104,13 @@ jobs:
     with:
       default_python: '3.12'
       submodules: false
-      coverage: codecov
+      coverage: ''
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply .github/workflows/ci.yml:107]: Does it matter for the cronjobs for line coverage?

File: tox.ini (hunk 1 above)
Comment (nabobalis) [inline tox.ini:19]:
```suggestion
```

**Comment by nabobalis** [comment]
The job will always fail, I cant work out how to stop it careing about the stability rst file.

File: docs/conf.py [hunk 4]
```diff
@@ -75,6 +75,7 @@
 sunpy.log.setLevel(ori_level)
 
 # For the linkcheck
+linkcheck_exclude_documents = ["reference/stability.rst"]
```
Comment (Cadair) [inline docs/conf.py:78]:
```suggestion
linkcheck_exclude_documents = ["stability.rst"]
```

File: .github/workflows/ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (nabobalis) [reply .github/workflows/ci.yml:106]: I will delete the line instead.

**Comment by nabobalis** [comment]
How I missed the linkcheck_exclude_documents I will never know.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -103,14 +103,13 @@ jobs:
...
```
Comment (nabobalis) [inline .github/workflows/ci.yml:106]:
```suggestion
```

**Comment by Cadair** [comment]
I can't make it work though lol

File: docs/conf.py [hunk 6]
```diff
@@ -362,7 +366,21 @@ def rstjinja(app, docname, source):
...
+def lol(app, docname, source):
+    """
+    lol
+    """
+    if app.builder.format == 'html':
+        return
+    files_to_remove = ["reference/stability", "dev_guide/index"]
+    for file in files_to_remove:
+        docname.found_docs.remove(file)
+        source.remove(file)
```
Comment (nabobalis) [inline docs/conf.py:378]: It was a moon shot and I misssed.

File: docs/conf.py [hunk 7]
```diff
@@ -362,7 +366,24 @@ def rstjinja(app, docname, source):
         source[0] = rendered
 
 
+def lol(app, docname, source):
```
Comment (wtbarnes) [inline docs/conf.py:369]: This function removes the unrendered parts of the jinja templates. I think the name should reflect that.

File: docs/dev_guide/contents/newcomers.rst [hunk 8]
```diff
@@ -84,7 +84,7 @@ If you find an issue you'd like to work on, please make sure to add a comment to
     This applies to everyone who wants to contribute during work time no matter who their employer is.
     You should start by checking if there is a Open Source Software Policy (e.g., `Standford's policy <https://otl.stanford.edu/open-source-stanford>`__) for your work place.
     If not, `OSS-Watch <http://oss-watch.ac.uk/resources/contributing>`__ summaries what you will need to check and who to ask, however this resource is aimed at a UK readers.
-    As an example, `Standford's guidance <https://otl.stanford.edu/sites/g/files/sbiybj10286/f/otlcopyrightguide.pdf>`__ allows someone to contribute and open source their code.
+    As an example, `Standford's guidance <https://otl.stanford.edu/software>`__ allows someone to contribute and open source their code.
```
Comment (wtbarnes) [inline docs/dev_guide/contents/newcomers.rst:87]:
```suggestion
    As an example, `Stanford's guidance <https://otl.stanford.edu/software>`__ allows someone to contribute and open source their code.
```

File: docs/conf.py [hunk 9]
```diff
@@ -362,7 +366,24 @@ def rstjinja(app, docname, source):
...
+    """
+    lol
```
Comment (ayshih) [inline docs/conf.py:371]: I'm not going to object on the function name, but there needs to be a docstring that explains this all

**Review by ayshih — CHANGES_REQUESTED** [review]

File: docs/conf.py (hunk 7 above)
Comment (wtbarnes) [reply docs/conf.py:369]: And should include a description

**Review by wtbarnes — CHANGES_REQUESTED** [review]

**Comment by wtbarnes** [comment]
I think the test fail is real. It looks like the numpy strings and floats are not being rendered as expected as strings.

**Comment by nabobalis** [comment]
> I think the test fail is real. It looks like the numpy strings and floats are not being rendered as expected as strings.

You mean the online tests? Yes that is known, something is downgrading numpy to <2 and breaking those tests.

File: docs/dev_guide/contents/newcomers.rst [hunk 10]
```diff
@@ -82,9 +82,9 @@ If you find an issue you'd like to work on, please make sure to add a comment to
     sunpy is Free and open-source software (FOSS), under the BSD-2 license. By contributing you are stating that you have the right to and agree to have your work distributed under the terms of this license.
 
     This applies to everyone who wants to contribute during work time no matter who their employer is.
-    You should start by checking if there is a Open Source Software Policy (e.g., `Standford's policy <https://otl.stanford.edu/open-source-stanford>`__) for your work place.
+    You should start by checking if there is a Open Source Software Policy (e.g., `Stanford's policy <https://otl.stanford.edu/open-source-stanford>`__) for your work place.
     If not, `OSS-Watch <http://oss-watch.ac.uk/resources/contributing>`__ summaries what you will need to check and who to ask, however this resource is aimed at a UK readers.
```
Comment (alasdairwilson) [inline docs/dev_guide/contents/newcomers.rst:86]:
```suggestion
    If not, `OSS-Watch <http://oss-watch.ac.uk/resources/contributing>`__ summarises what you will need to check and who to ask, however this resource is aimed at a UK readers.
```

File: sunpy/image/resample.py [hunk 11]
```diff
@@ -135,7 +135,7 @@ def _resample_spline(orig, dimensions, offset, m1):
 
 def reshape_image_to_4d_superpixel(img, dimensions, offset):
     """
-    Re-shape the two dimension input image into a a four dimensional array
+    Re-shape the two dimension input image into a four dimensional array
```
Comment (alasdairwilson) [inline sunpy/image/resample.py:138]:
```suggestion
    Re-shape the two-dimensional input image into a four-dimensional array
```

**Review by alasdairwilson — APPROVED** [review]
This seems fine for me now.

File: sunpy/net/jsoc/jsoc.py [hunk 12]
```diff
@@ -68,14 +68,14 @@ class JSOCClient(BaseClient):
 
     Notes
     -----
-    The full list of ``Series`` is available through this `site <http://jsoc.stanford.edu>`_.
+    The full list of ``Series`` is available through this `site <http://jsoc.stanford.edu>`__.
```
Comment (alasdairwilson) [inline sunpy/net/jsoc/jsoc.py:71]: Did you want to link direct to the page here?
```suggestion
    The full list of ``Series`` is available via ` this page <http://jsoc.stanford.edu/JsocSeries_DataProducts_map.html>`__.
```

**Review by alasdairwilson — APPROVED** [review]

File: examples/units_and_coordinates/spice.py [hunk 13]
```diff
@@ -12,7 +12,7 @@
 examples of mission SPICE kernels:
 
 * `Solar Orbiter <https://www.cosmos.esa.int/web/spice/solar_orbiter>`__
-* `Parker Solar Probe <https://sppgway.jhuapl.edu/ancil_products>`__
+* `Parker Solar Probe <https://psp-gateway.jhuapl.edu/>`__
```
Comment (ayshih) [inline examples/units_and_coordinates/spice.py:15]:
```suggestion
* `Parker Solar Probe <https://psp-gateway.jhuapl.edu/website/Ancillary/SclkFiles>`__
```

**Review by ayshih — APPROVED** [review]
Just one suggestion

**Review by wtbarnes — APPROVED** [review]

**Pull request merged**